fail2ban/server/filter.py

```diff
@@ -844,35 +943,50 @@ def getJournalMatch(self, match): # pragma: no cover - Base class, not used
 
 import socket
 import struct
+from .utils import Utils
 
 
 class DNSUtils:
 
 	IP_CRE = re.compile("^(?:\d{1,3}\.){3}\d{1,3}$")
 
+	# todo: make configurable the expired time and max count of cache entries:
+	CACHE_dnsToIp = Utils.Cache(maxCount=1000, maxTime=60*60)
```

Comment:
I would say that an hour for DNS cache expiration is a bit too big... what about  at most 5*60? 
just a comment: anyways -- in "real world" majority of filters should avoid actions based on logged provided by client domain names, thus resolving names back to possibly victim IPs and admins should adjust (where needed) their services to log IPs.